source/reference/logging.txt

```diff
@@ -0,0 +1,202 @@
+
+To output log messages to file, start ``mongosync`` with the 
+:setting:`logPath` setting or :option:`--logPath` option.
+
+When logging to file, ``mongosync`` writes logs to a file in the configured
+directory.  The directory is created if it does not exist.  If a ``mongosync.log``
+file already exists in the directory, ``mongosync`` rotates the log files to
+preserve the old logs as it begins a new file.
+
+.. code-block:: console
+
+   $ ls /var/log/mongosync
+
+For example: 
+
+.. code-block:: none
+   :copyable: false
+
+   mongosync-2022-06-17T16-27-58.187.log  mongosync.log
+   mongosync-2022-06-21T13-31-42.668.log
+
+
+Verbosity
+---------
+
+{+c2c-product-name+} supports user-defined verbosity to increase or decrease
+the level of log messages :program:`mongosync` outputs.
+
+The verbosity level can be set using the :option:`--verbosity` option from the
+command-line or the :setting:`verbosity` setting in the configuration file.
+
+Format
+------
+
+:program:`mongosync` outputs log messages in structured JSON format. Log messages
+are written as a series of key-value pairs, where each key indicates a log message
+field type, such as the severity level, log component, and timestamp.
+
+For example:
+
+.. code-block:: json
+   :copyable: false
+
+   {
+        "level": "info",
+        "mongosyncID": "shard01",
+        "component": "Change Event Application",
+        "time": "2022-06-21T09:31:42-04:00",
+        "message": "Starting change stream reader."
+   }
+
+Components
+----------
+
+Log entries use the ``componentName`` field to indicate the component that generated
+the log message.
+
+For example:
+
+.. code-block:: json
+   :copyable: false
+
+   {
+      "level":         "info",
+      "serverID":      "0624a3e",
+      "mongosyncID":   "coordinator",
+      "componentName": "CHECKPOINT",
+      "time":          "2022-07-13T17:44:09.837830382-04:00",
+      "message":       "starting Mongosync from change event application phase."
+   }
+
+``mongosync`` uses the following components:
+
+.. data:: Change Event Application
+
+   Messages relate to the Change Event Application (CEA), which monitors a
+   change stream on the source cluster and converts :ref:`change events
+   <change-events>` into writes.
+
+
+.. data:: Change Event dispatcher
+
+   Messages relate to the Change Event Dispatcher, which batches change stream
+   events from the source cluster.
+
+
+.. data:: Change Event processor
+
+   Messages relate to the Change Event Processor, which receives change stream
+   events and applies them to the destination cluster.
+
+
+.. data:: Change stream reader
+
+   Messages relate to the Change Stream Reader, which reads change stream events
+   on the source cluster and sends them to a channel for processing.
+
+
+.. data:: CHECKPOINT
+
+   Messages relate to snapshots taken before and after an operation.
+
+
+.. data:: Collection Copy
+
+   Messages relate to the copying of a collection from the source to destination
```

Comment:
Consider:
relate to copying a collection

Reply:
Fixed